js/src/product-feed/index.test.js

```diff
@@ -0,0 +1,180 @@
+			getQuery.mockImplementation( () => {
+				return {
+					guide: GUIDE_NAMES.SUBMISSION_SUCCESS,
+				};
+			} );
+		} );
+
+		afterAll( () => {
+			getQuery.mockReset();
+		} );
+
+		test( 'Should render SubmissionSuccessGuide', () => {
+			const { queryByText } = render( <ProductFeed /> );
+			expect(
+				queryByText( SUBMISSION_SUCCESS_GUIDE_TEXT )
+			).toBeInTheDocument();
+		} );
+
+		test( 'Should not render CustomerEffortScorePrompt', () => {
+			const { queryByText } = render( <ProductFeed /> );
+			expect( queryByText( CES_PROMPT_TEXT ) ).not.toBeInTheDocument();
+		} );
+	} );
+
+	describe( `When the query string "guide" does not equals to ${ GUIDE_NAMES.SUBMISSION_SUCCESS }`, () => {
+		beforeAll( () => {
+			getQuery.mockImplementation( () => {
+				return {};
+			} );
+		} );
+
+		afterAll( () => {
+			getQuery.mockReset();
+		} );
+
+		test( 'Should not render SubmissionSuccessGuide', () => {
+			const { queryByText } = render( <ProductFeed /> );
+			expect(
+				queryByText( SUBMISSION_SUCCESS_GUIDE_TEXT )
+			).not.toBeInTheDocument();
+		} );
+
+		describe( 'And both canCESPromptOpenLocal and wcTracksEnabled are false', () => {
+			beforeAll( () => {
+				localStorage.get.mockImplementation( () => {
+					return false;
+				} );
+				isWCTracksEnabled.mockImplementation( () => {
+					return false;
+				} );
+			} );
+
+			afterAll( () => {
+				localStorage.get.mockReset();
+				isWCTracksEnabled.mockReset();
+			} );
+
+			test( 'Should not render CustomerEffortScorePrompt', () => {
+				const { queryByText } = render( <ProductFeed /> );
+				expect(
+					queryByText( CES_PROMPT_TEXT )
+				).not.toBeInTheDocument();
+			} );
+		} );
+
+		describe( 'And canCESPromptOpenLocal is true but wcTracksEnabled is false', () => {
+			beforeAll( () => {
+				localStorage.get.mockImplementation( () => {
+					return true;
+				} );
+				isWCTracksEnabled.mockImplementation( () => {
+					return false;
+				} );
+			} );
+
+			afterAll( () => {
+				localStorage.get.mockReset();
+				isWCTracksEnabled.mockReset();
+			} );
+
+			test( 'Should not render CustomerEffortScorePrompt', () => {
+				const { queryByText } = render( <ProductFeed /> );
+				expect(
+					queryByText( CES_PROMPT_TEXT )
+				).not.toBeInTheDocument();
+			} );
+		} );
+
+		describe( 'And wcTracksEnabled is true but canCESPromptOpenLocal is false', () => {
+			beforeAll( () => {
+				localStorage.get.mockImplementation( () => {
+					return false;
+				} );
+				isWCTracksEnabled.mockImplementation( () => {
+					return true;
+				} );
+			} );
+
+			afterAll( () => {
+				localStorage.get.mockReset();
+				isWCTracksEnabled.mockReset();
+			} );
+
+			test( 'Should not render CustomerEffortScorePrompt', () => {
+				const { queryByText } = render( <ProductFeed /> );
+				expect(
+					queryByText( CES_PROMPT_TEXT )
+				).not.toBeInTheDocument();
+			} );
+		} );
+
+		describe( 'And both canCESPromptOpenLocal and wcTracksEnabled are true', () => {
```

Comment:
💅 

RE-  comments regarding `isWCTracksEnabled` 

It will also simplify the test. We can have a separated test for the CESPrompt checking that and here only test LocalStorage / SubmissionSuccessGuide